plugins/woocommerce/tests/php/includes/shipping/flat-rate/class-wc-shipping-flat-rate-test.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+// phpcs:disable Squiz.Classes.ClassFileName.NoMatch, Squiz.Classes.ValidClassName.NotCamelCaps -- backcompat nomenclature.
+
+/**
+ * Test for WC_Shipping_Flat_Rate class.
+ */
+class WC_Shipping_Flat_Rate_Test extends WC_Unit_Test_Case {
+
+	/**
+	 * @var WC_Shipping_Flat_Rate Shipping method instance.
+	 */
+	private $sut;
+
+	/**
+	 * @var Closure Function to call protected method evaluate_cost.
+	 */
+	private $call_evaluate_cost;
+
+	/**
+	 * Set up test case.
+	 *
+	 * @return void
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		$this->sut                = new WC_Shipping_Flat_Rate();
+		$this->call_evaluate_cost = function ( $sum, $args ) {
+			return $this->evaluate_cost( $sum, $args );
+		};
+		update_option( 'woocommerce_price_decimal_sep', ',' );
+		update_option( 'woocommerce_price_thousand_sep', '.' );
```

Comment:
[Non-blocking observation] I'm somewhat stuck on "we can assert that it also works vice-versa" or if we should move this setup to the individual tests that need them. That way we can test both `.` and `,` as the different separators (think `@dataProvider`).

E.g. if/when we add thousand separators and remove the `markTestSkipped`:

```php
$this->markTestSkipped( 'This test currently fails because we dont support thousand seperator in shipping price.' );
```

Are we expecting this to be a full test-suite? In that case, we should test that both `.` and `,` works as expected as thousand separators.
My PoV: then these tests are here to communicate that it's not supported and we're not intending to create full coverage tests now since lots of things can change in the future that would make them outdated anyway.

Reply:
Yeah, in general, more tests around this functionality would be nice, not just for flat shipping but around other types also. It's a miss that we already don't have extensive coverage. For this PR I wanted to cover the cases that may break, and the cases the PR was trying t fix, as a way of documentation. Let's add more coverage in separate non fix targeting PRs.